src/plugins/discover/public/application/angular/discover.js

```diff
@@ -979,20 +707,11 @@ function discoverController($element, $route, $scope, $timeout, $window, Promise
   };
 
   $scope.updateDataSource = () => {
-    const { indexPattern, searchSource } = $scope;
-    searchSource
-      .setField('index', $scope.indexPattern)
-      .setField('size', $scope.opts.sampleSize)
-      .setField(
-        'sort',
-        getSortForSearchSource(
-          $scope.state.sort,
-          indexPattern,
-          config.get(SORT_DEFAULT_ORDER_SETTING)
-        )
-      )
-      .setField('query', data.query.queryString.getQuery() || null)
-      .setField('filter', filterManager.getFilters());
+    updateSearchSource($scope.searchSource, {
```

Comment:
This PR in general will cause trouble with any work that's being done in parallel, most definitely work on supporting Fields API. We should coordinate how and when we want to merge this.

Reply:
Well, it should not block you from anything, so I'm good with waiting to merge , not to cause any troubles . will split up such work in smaller PRs in the future